src/components/search.js

```diff
@@ -1,86 +1,88 @@
 /**
  * Search
+ * @flow
  */
 
 import React from 'react'
 import debounce from 'debounce'
-import Button from './ui/button'
-import UniqueID from '../mixins/uniqueId'
 import DataList from './datalist'
-import createClass from 'create-react-class'
-import { func } from 'prop-types'
-
-let Search = createClass({
-  mixins: [UniqueID],
-
-  statics: {
-    // The minimum number of characters before searching
-    THRESHOLD: 2,
-    // The minimum time between change events
-    INTERVAL: 150
-  },
-
-  propTypes: {
-    onChange: func.isRequired
-  },
-
-  getDefaultProps() {
-    return {
-      datalist: []
-    }
-  },
+import { type Record } from '../record'
 
-  getInitialState() {
-    return {
-      search: '',
-      debouncedChange: debounce(this.props.onChange, Search.INTERVAL)
-    }
-  },
+type Props = {
+  datalist: Record[],
+  onChange: string => *
+}
+
+type State = {
+  search: string
+}
+
+let uid = 0
+
+// The minimum number of characters before searching
+const THRESHOLD = 2
+
+// The minimum time between change events
+const INTERVAL = 150
+
+export default class Search extends React.Component<Props, State> {
+  id: number
+  debouncedChange: string => *
+
+  state = {
+    search: ''
+  }
+
+  constructor(props: Props) {
+    super(...arguments)
+
+    this.id = uid++
+    this.debouncedChange = debounce(props.onChange, INTERVAL)
+  }
 
   render() {
-    let id = 'ars_search_' + this.state.id
-    let list = 'ars_search_list' + this.state.id
+    let id = `ars_search_${this.id}`
+    let list = `ars_search_list_${this.id}`
 
     return (
-      <form className="ars-search" onSubmit={this._onSubmit}>
+      <form className="ars-search" onSubmit={this._onSubmit.bind(this)}>
         <label className="ars-search-label" htmlFor={id}>
           Search
         </label>
         <input
           id={id}
           list={list}
-          ref="input"
           type="search"
           className="ars-search-input"
-          onChange={this._onChange}
           placeholder="Search"
-          onKeyUp={this._onKeyUp}
           value={this.state.search}
+          onChange={this._onChange.bind(this)}
+          onKeyUp={this._onKeyUp.bind(this)}
         />
         <DataList id={list} items={this.props.datalist} />
-        <Button className="ars-hidden">Submit</Button>
+        <button className="ars-hidden">Submit</button>
```

Comment:
This feels like more than modernizing and changing how it works. Is all `<Button>` doing is just wrapping a `<button>` and passing the className along and literally nothing else?

Reply:
Ah I guess so :). The `<Button>` component adds an `<Ink />` instance to get the click/hold effect (https://react-ink.netlify.com/).

This button is hidden, so I turned it into a regular button because it doesn't need the special ars arsenal classes, or the Ink component. Would you rather see this in a separate PR?

Reply:
Maybe not worth it. It was hidden anyway, it's still a real button underneath, and doesn't really change how it works like I thought at first.